hbase_storage/src/main/java/com/splicemachine/storage/HNIOFileSystem.java

```diff
@@ -187,58 +172,62 @@ public void concat(Path target, Path... sources)  throws IOException {
-                aclS = new AclStatus.Builder().owner("unknown").group("unknown").build();
-            } catch(Exception e){
-                e = exceptionFactory.processRemoteException(e); //strip any multi-retry errors out
-                //noinspection ConstantConditions
-                if(e instanceof UnsupportedOperationException|| e instanceof AclException){
-                    /*
-                     * Some Filesystems don't support aclStatus. In that case,
-                     * we replace it with our own ACL status object
-                     */
-                    aclS=new AclStatus.Builder().owner("unknown").group("unknown").build();
-                }else{
-                    /*
-                     * the remaining errors are of the category of FileNotFound,UnresolvedPath,
-                     * etc. These are environmental, so we should throw up here.
-                     */
-                    throw new IOException(e);
-                }
-            }
-            this.aclStatus = aclS;
-            boolean readable;
-            try{
-                fs.access(path,FsAction.READ);
-                readable= true;
-            }catch(IOException ioe){
-               readable = false;
+            try {
+                this.fileStatus = fs.getFileStatus(path);
+            } catch( FileNotFoundException e )
+            {
+                this.fileStatus = null;
             }
-            boolean writable;
-            try{
-                fs.access(path,FsAction.WRITE);
-                writable = true;
-            }catch(IOException ioe){
-                writable = false;
+        }
+
+        // these two methods are to avoid having to re-calculate the list of files in the directory
+        // for isEmptyDirectory
+        // todo(martinrupp): replace with recursive listdir
+        private FileStatus rootFileStatusArr[];
+        private FileStatus[] listRoot() throws IOException {
+            if (rootFileStatusArr != null || !exists() || !fileStatus.isDirectory()) return rootFileStatusArr;
+            rootFileStatusArr = fs.listStatus(path);
+            return rootFileStatusArr;
+        }
+        // note this is expensive for deeply nested directories. avoid calling fileCount, spaceConsumed and size
+        // partly copied FileSystem.getContentSummary , however with cached fileStatus and using (cached) listRoot for listing root.
+        private ContentSummary getContentSummary() {
+            if( contentSummary != null ) return contentSummary;
+            try {
+                if (fileStatus.isFile()) {
+                    // f is a file
+                    long length = fileStatus.getLen();
+                    contentSummary = new ContentSummary.Builder().length(length).
+                            fileCount(1).directoryCount(0).spaceConsumed(length).build();
+                }
+                else {
+                    // f is a directory
+                    long[] summary = {0, 0, 1};
```

Comment:
Nitpick: explicit variables would be better IMHO (e.g. length / fileCount / dirCount)

Reply:
this is copied from FileSystem.java, will fix.